packages/proxy/lib/http/util/prerequests.ts

```diff
@@ -55,59 +97,63 @@ export class PreRequests {
     this.sweepInterval = setInterval(() => {
       const now = Date.now()
 
-      Object.entries(this.prerequestTimestamps).forEach(([key, timestamp]) => {
+      this.pendingPreRequests.removeMatching(({ timestamp, browserPreRequest }) => {
         if (timestamp + this.requestTimeout * 2 < now) {
-          debugVerbose('timed out unmatched pre-request %s: %o', key, this.pendingPreRequests[key])
+          debugVerbose('timed out unmatched pre-request: %o', browserPreRequest)
           metrics.unmatchedPreRequests++
```

Comment:
Is this metric still valid?  Should we only be increasing it if we actually remove the stack matching the key?